src/Auth0.Core/Exceptions/ErrorApiException.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Net;
+
+namespace Auth0.Core.Exceptions
+{
+    /// <summary>
+    /// Represents errors that occur when making API calls.
+    /// </summary>
+    public class ErrorApiException : ApiException
```

Comment:
So this would be used for all errors besides a rate limit encountered? And then we could add additional exceptions for other errors, if we're able to do something helpful?

Reply:
Yep.  Previously there was only one exception type which had ApiError.  Now we have RateLimitException which won't have that so I moved ApiError to another new sub-type of ApiException.

This is the safest path for migration - if they were using ApiError they'll just need to switch their catch from ApiException to ErrorApiException when they see the problem in the compiler.

If I was to leave the existing ApiException as-is then RateLimitException would be not handled and they could instead see exceptions in production they previously handled when those 429 errors were just plain ApiException.